src/services/services.ts

```diff
@@ -5008,6 +5015,94 @@ namespace ts {
+            if (node.parent.kind === SyntaxKind.ShorthandPropertyAssignment) {
+                const result: ReferenceEntry[] = [];
+                getReferenceEntryForShorthandPropertyAssignment(node, typeChecker, result);
+                return result.length > 0 ? result : undefined;
+            }
+            else if (node.kind === SyntaxKind.SuperKeyword || isSuperProperty(node.parent)) {
+                // References to and accesses on the super keyword only have one possible implementation, so no
+                // need to "Find all References"
+                const symbol = typeChecker.getSymbolAtLocation(node);
+                if (symbol.valueDeclaration) {
+                    return [getReferenceEntryFromNode(symbol.valueDeclaration)];
+                }
+            }
+            else {
+                const entries: ImplementationLocation[] = [];
+                // Perform "Find all References" and retrieve only those that are implementations
+                const result = getReferencedSymbolsForNode(node, program.getSourceFiles(), /*findInStrings*/false, /*findInComments*/false, /*implementations*/true);
+
+                forEach(result, referencedSymbol => {
+                    if (referencedSymbol.references) {
+                        forEach(referencedSymbol.references, entry => {
+                            entries.push({
+                                textSpan: entry.textSpan,
+                                fileName: entry.fileName
+                            });
+                        });
+                    }
+                });
+                return entries;
+            }
+        }
+
+        function getReferenceEntryForShorthandPropertyAssignment(node: Node, typeChecker: TypeChecker, result: ReferenceEntry[]): void {
+            const refSymbol = typeChecker.getSymbolAtLocation(node);
+            const shorthandSymbol = typeChecker.getShorthandAssignmentValueSymbol(refSymbol.valueDeclaration);
+
+            if (shorthandSymbol) {
+                for (const declaration of shorthandSymbol.getDeclarations()) {
+                    if (getMeaningFromDeclaration(declaration) & SemanticMeaning.Value) {
+                        result.push(getReferenceEntryFromNode(declaration));
+                    }
+                }
+            }
+        }
+
+        function isNameOfImplementation(node: Node): boolean {
+            const parent = node.parent;
+
+            if (isDeclarationName(node)) {
+                if (isVariableLike(parent)) {
+                    if (parent.kind === SyntaxKind.VariableDeclaration) {
+                        const parentStatement = parent.parent && parent.parent.parent;
+                        if (parentStatement && hasModifier(parentStatement, ModifierFlags.Ambient)) {
+                            return true;
+                        }
+                    }
+                    return !!parent.initializer;

```

Comment:
I would put this (`if (parent.initializer) { return true; }`) first since it's the usual case.